### DISCUSSION TITLE
Additional RSS data?

### DISCUSSION THREAD

**Discussion by eric-hc** [facebook/docusaurus#5623] [Q&A]
Is it possible to add additional RSS attributes for blog items? For example, given this item,

```xml
<item>
    <title>
        <![CDATA[ Google APIs: authentication with TypeScript ]]>
    </title>
    <link>https://blog.johnnyreilly.com/2021/09/10/google-apis-authentication-with-typescript</link>
    <guid>Google APIs: authentication with TypeScript</guid>
    <pubDate>Fri, 10 Sep 2021 00:00:00 GMT</pubDate>
    <description>
        <![CDATA[ Google has a wealth of APIs which we can interact with. At the time of writing, there's more than two hundred         available; including YouTube, Google Calendar and GMail (alongside many others). To integrate with these APIs, it's necessary to authenticate and then use that credential with the API. This post will take you through how to do just that using TypeScript. It will also demonstrate how to use one of those APIs: the Google Calendar API. ]]>
    </description>
</item>
```

Could I add an additional piece of metadata?

**Comment by slorber** [comment]
Can you explain what piece of metadata you want to add exactly, and what API you'd like for this?

Is this an std feature supported by RSS? What about other feed types like Atom?

What's the use-case for this extra metadata?

Our lib's typing is here: https://github.com/jpmonette/feed/blob/master/src/typings/index.ts
There's a concept of "extension" allowing an `any` data structure, maybe that's the way to do it?

**Comment by filiptammergard** [comment]
I looking for ways of getting the blog post tags through the RSS feed, to be able to show the tag in the feed summary. Using the extension concept would be fine for me!

Is this doable?

**Reply by slorber** [reply]
@filiptammergard my goal is to actually figure out if this extension concept should be used for this use case. 

Do you have any reference link or blog post that mentions adding blog post tags to an RSS feed? 

Technically this seems doable, the question is more: is it the right way to solve this problem.

**Reply by filiptammergard** [reply]
After looking around, I'd say there actually is a tag exactly for this use case: `<category>`. 

> Used with both the <item> and <channel> tags to specify a category for an entire RSS feed or for individual RSS items.

Here are a couple of examples: http://www.landofcode.com/rss-reference/category-tag.php

What do you think @slorber? 

Is this something I could help with by submitting a PR?

**Reply by slorber** [reply]
Thanks @filiptammergard that seems like a good idea.

I didn't test but in blogUtils.ts it should probably be:


```ts
  blogPosts.forEach((post) => {
    const {
      id,
      metadata: {title: metadataTitle, permalink, date, description, authors},
    } = post;
    feed.addItem({
      title: metadataTitle,
      id,
      link: normalizeUrl([siteUrl, permalink]),
      date,
      description,
      author: authors.map(toFeedAuthor),
      category: post.metadata.tags.map((tag) => ({
        name: tag.label,
        domain: normalizeUrl([blogBaseUrl, tag.permalink]),
      })),
    });
  });
```

@eric-hc is this solving your use-case or you have a totally different need?

**Reply by slorber** [reply]
Note, there's also a `feed.addCategory("string")`, it's not very clear to me when it should be used

**Reply by filiptammergard** [reply]
Great @slorber! Before givning this a go I have to ask:

Our use case is that we're showing a summary of the latest blog posts from /blog on the landing page. I've read the docs carefully, but using the RSS feed was the easiest way I could find. But is there a way to get data through the blog plugin directly that's already in place? Or some other way?

**Reply by slorber** [reply]
@filiptammergard I'd like in the future to have a declarative system so that any page can "request" the blog data module (somehow similar to Gatsby but without GraphQL).

Another possible solution today is to "extend" the blog plugin to make it create your homepage:
- https://github.com/facebook/docusaurus/discussions/4800
- https://github.com/facebook/docusaurus/issues/4138

We also have a blog archive page, you could choose your own component to render it in config and configure it to be at `/` (also requires the whole blog to be at `/`)
